content/purgatorio/launch-and-mantain-the-system/automation-and-reproducibility.md

```diff
@@ -1,5 +1,242 @@
+
+::: warning
+Often AutoMl jobs consume a lot of computational resources (since it often requires a search in the model architectures space), so watch your back with the billing of the platform that you choose!
+:::
+
+## No Free Lunch and Black Boxes
+
+It may seem too good to be true, right? There are actually some aspects to be careful about when choosing an AutoML approach. 
+
+- Not all data types are suitable / supported by the currently available frameworks (autoML works best with tabular data). More complex data formats (like images, videos, audio...) aren't often considered in modern AutoML frameworks.
+- However, it is important to know the data well, to know what information (often called signals) it contains, and to know what information we want to derive from it. 
+- There is no such thing (as usual) as a _silver bullet_, so don't expect spectacular results in a magical way, on every problem you approach with AutoML. 
+- It's usually computationally intensive, this can rapidly inflate the billing costs of the platform you're working on.  
+
+Another **issue** you must consider when applying AutoML techniques, is the **opacity** that you bring to the process.
+
+The concept of "[**Black Box**](https://towardsdatascience.com/the-black-box-metaphor-in-machine-learning-4e57a3a1d2b0)", linked to more complex ML algorithms (such as neural networks) makes the resulting ML models "opaque" and therefore difficult to explain, **should be equally considered for all steps of the Data Science process to which some automatic decision applies**.
+
+For example:
+
+- If some data preprocessing steps are decided from the machine, _why_ are they like that?
+- Is it possible that these choices will affect the models that will then be trained on the data? If yes, _how_?
+- If some features have been dropped and others have been chosen for the training, _why_? 
+- If a certain model was chosen to be trained on data, _why_ that model? And if the data changed over time, would that model be able to _adapt_ to the new situation (after appropriate re-training)?
+
+As you can see, as there are many pros in using AutoML, there are also various issues that can be problematic.
+
+::: tip
+**What does Virgiloo think of AutoML?**
+
+Basically, it's a very good set of tools, which when used in a smart way can make Data Scientist's life a lot easier, guiding him in the exploration and transformation of data, in the choice of the most suitable ML models and so on. 
+
+On the other hand, it injects opacity into every step of the Data Science process to which it is applied, so watch your back!
+::: 
+
+## Conclusions
+
+In this guide, you have seen how automation can bring enormous benefits to the whole Data Science process, especially in terms of the credibility of results (Reproducibility) and time savings. 
+
+You've also seen what AutoML is and what its (great) potential is, but also the potential risks it brings with it in terms of opacity (Black Box concept). 
```

Comment:
It'll be a good idea to have a resources section and add a bunch of the links from the here. including links like this:

- [Things to know: AutoML](https://github.com/neomatrix369/awesome-ai-ml-dl/blob/master/things-to-know.md#automated-machine-learning-tools-or-partial-automl-tools)
- [Cloud, Infra, DevOps](https://github.com/neomatrix369/awesome-ai-ml-dl/blob/master/cloud-devops-infra/README.md#cloud-devops-infra)

Let me know if you need any more related ones.